packages/turbo-workspaces/src/utils.ts

```diff
@@ -205,6 +201,47 @@ function getMainStep({
   } ${action === "remove" ? "from" : "to"} ${project.name}`;
 }
 
+/**
+ * At the time of writing, bun only support simple globs (can only end in /*) for workspaces. This means we can't convert all projects
+ * from other package manager workspaces to bun workspaces, we first have to validate that the globs are compatible.
+ *
+ * NOTE: It's possible a project could work with bun workspaces, but just not in the way it is currently defined. We will
+ * not change existing globs to make a project work with bun, we will only convert projects that are already compatible.
+ *
+ * This function matches the behavior of bun's glob validation: https://github.com/oven-sh/bun/blob/main/src/install/lockfile.zig#L2889
```

Comment:
Good idea to port this precisely. Use a GitHub permalink (`y` on your keyboard after going to that URL) for this. (This link is already broken.)

The other package managers arbitrarily switch between micromatch and minimatch, but fortunately that's basically compatible except for cases that are _far_ too complicated for workspace definition.

Personal opinion: these shouldn't have allowed globs to begin with.

Reply:
yes I suggest using a permalink. note that eventually we will implement a glob in zig and this limitation will go away. A bunch of companies are asking for it so we might use a [filesystem-only libc implementation](https://man7.org/linux/man-pages/man3/glob.3.html) in the worst case